pkg/otlp/model/translator/metrics_translator.go

```diff
@@ -33,6 +34,14 @@ import (
 
 const metricName string = "metric name"
 
+var _ source.Provider = (*noSourceProvider)(nil)
+
+type noSourceProvider struct{}
```

Comment:
Not entirely clear what this is. Is it supposed to be a sourcer which doesn't know the sourcer? If so, maybe `unknownSourcer` (or `unknownSourceProvider` if you don't accept my interface name change suggestion).